### ISSUE TITLE
Compile errors with Haxe 3.3.0-rc1

### ISSUE THREAD

**Issue by mikedotalmond** [zweimal/HxSignal#3]
I get the following errors when attempting to build with Haxe 3.3.0-rc1 (messages edited to remove duplicates).

`hxsignal/0,9,1/hxsignal/ds/LinkedList.hx:42: characters 7-16 : Array<Dynamic> should be _List.ListNode<hxsignal.ds.LinkedList.T>`

`hxsignal/0,9,1/hxsignal/ds/LinkedList.hx:64: characters 11-15 : Array access is not allowed on _List.ListNode<hxsignal.ds.LinkedList.T>`

`hxsignal/0,9,1/hxsignal/Signal.hx:34: lines 34-65 : @:genericBuild cannot be used in macros`

**Comment by zweimal** [comment]
Thanks, Mike.
It looks like a deep change in last version of haxe, so I have to do some research on how to make this work again.

**Comment by mikedotalmond** [comment]
Thanks for looking into it.

Yeah, I had a quick look to see if it was something I could easily resolve, but does look like it's releated to some [breaking changes in Haxe 3.3](https://github.com/HaxeFoundation/haxe/wiki/Breaking-changes-in-Haxe-3.3.0), and I didn't feel I knew hxSignal's internals well enough to look much further.

**Comment by bmfs** [comment]
I've got a branch with those fixes. I'll create a pull request with it later this day.

The "@:genericBuild cannot be used in macros" is a matter of wrapping the @:genericBuild call with  ```#if !macro ...  #end```.

The issue with the LinkedList is due to a change on the linked list node data structure.
Before, a node was an Array<Dynamic> and now it's a ListNode.

Unfortunately the ListNode is declared as a private class.

**Comment by zweimal** [comment]
So, there are 2 problems. That's what you mean?

**Comment by bmfs** [comment]
Yes.

As @mikedotalmond  mentioned, the errors are the following

```
hxsignal/0,9,1/hxsignal/ds/LinkedList.hx:42: characters 7-16 : Array<Dynamic> should be _List.ListNode<hxsignal.ds.LinkedList.T>

hxsignal/0,9,1/hxsignal/ds/LinkedList.hx:64: characters 11-15 : Array access is not allowed on _List.ListNode<hxsignal.ds.LinkedList.T>

hxsignal/0,9,1/hxsignal/Signal.hx:34: lines 34-65 : @:genericBuild cannot be used in macros
```

**Comment by zweimal** [comment]
Sorry, I didn't undetstand how to solve the macro issue.
LinkedList needs a refactor regarding you said.

**Comment by bmfs** [comment]
Check my quick fix for those issues.
https://github.com/bmfs/HxSignal/tree/fix_linked_list_nodes

About the LinkNode, I've tried to use the @:access metadata, but it doesn't seem to work.

**Comment by zweimal** [comment]
Fixed through [#09f5ad2] (https://github.com/zweimal/HxSignal/commit/09f5ad2d705a1eb8ce8e63ee92204a7e9425c99b) and [#393bc43](https://github.com/zweimal/HxSignal/commit/393bc43bb11bf75b880feebb8fe800a9de59ba08)
